docs/settings.rst

```diff
@@ -0,0 +1,205 @@
+
+login_redirection_requires_https
+***********************
+
+This setting configures if dynamic login redirection URI must have the ``https`` scheme.
+
+login_uris_redirect_allowed_hosts
+**********************
+
+This setting configures the list of allowed host in dynamic URI redirections.
+
+Cache
+=====
+
+oidc_cache_provider_metadata
+****************************
+
+**Default** : ``False``
+
+When this setting is enabled, we will cache the calls to the autoconfiguration endpoint of the OIDC
+identity provider.
+
+oidc_cache_provider_metadata_ttl
+********************************
+
+**Default** : ``120``
+
+
+This settings has no effect if ``oidc_cache_provider_metadata`` is disabled.
+
+Otherwise, it configures the lifetime (in seconds) of cached response for the autoconfiguration of
+the identity provider.
+
+cache_django_backend
+********************
+
+This setting configures the cache backend that is used to store OIDC sessions details. It should be
+the name of a cache defined in the ``CACHES` django settings.
+You can read more about *Cache Management* :ref:`here <Cache Management>`.
+
+Hook
+====
+
+Hook settings are path to a python function that should be called in specific context. We use a custom syntax to reference a function of a module.
+
+The syntax is : ``<module path>:<function name>``.
+
+
+So for example, if you were to have a module named ``oidc.py`` next to your project settings with a function called ``logout_callback`` you should use the string ``<your application root module>.oidc:logout_callback`` in your settings.
+
+.. note::
+    Hook settings work on a provider by provider basis, you can have different hook functions for each of your identity providers
+
+
+.. note::
+    All those settings are optional
+
+hook_user_logout
+****************
+
+Calls the provided function on user logout. The function is called if the logout is successful, but before redirecting the user.
+
+This function takes two arguments :
+
+1. a request instance :class:`django:django.http.HttpRequest`
+2. the request args sent to the sso server (missing the id_token_hint element)
+
+If the user was logged in, you can get the user using ``request.user``.
+
+hook_user_login
+****************
+
+Calls the provided function on user login. The functions is called if the login is successful.
+
+This function takes two arguments :
+
+1. a request instance :class:`django:django.http.HttpRequest`
+2. a user instance :class:`django.contrib.auth.models.User`
+
+Since the user wasn't logged in, it is not yet attached to the request instance at this stage. As such trying to access ``request.user`` will return an unauthenticated user.
+
+hook_get_user
+*************
+
+Calls the provided function on user login. It takes two arguments :
+
+* the user info token (a dictionary) from the identity provider
+* the id token
```

Comment:
Not sure, c'est pas plutôt une liste de tokens là?

Reply:
Ça c'est `hook_get_user` ;)